frigate/events.py

```diff
@@ -49,208 +51,62 @@ def should_create_clip(self, camera, event_data):
-                playlist_lines.append(
-                    f"inpoint {int(playlist_start-clip['start_time'])}"
-                )
-            # if this is the ending clip, add an outpoint
-            if clip["start_time"] + clip["duration"] > playlist_end:
-                playlist_lines.append(
-                    f"outpoint {int(playlist_end-clip['start_time'])}"
-                )
+            if recordings_count > 0:
+                return True
+            logger.debug(f"Missing recording for {camera} clip. Waiting...")
+            time.sleep(5)
 
-        clip_name = f"{camera}-{event_data['id']}"
-        ffmpeg_cmd = [
-            "ffmpeg",
-            "-y",
-            "-protocol_whitelist",
-            "pipe,file",
-            "-f",
-            "concat",
-            "-safe",
-            "0",
-            "-i",
-            "-",
-            "-c",
-            "copy",
-            "-movflags",
-            "+faststart",
-            f"{os.path.join(CLIPS_DIR, clip_name)}.mp4",
-        ]
-
-        p = sp.run(
-            ffmpeg_cmd,
-            input="\n".join(playlist_lines),
-            encoding="ascii",
-            capture_output=True,
+        logger.warning(
+            f"Unable to verify clip for {camera}. There were no recordings for this camera."
         )
-        if p.returncode != 0:
-            logger.error(p.stderr)
-            return False
-        return True
+        return False
 
     def run(self):
         while not self.stop_event.is_set():
             try:
                 event_type, camera, event_data = self.event_queue.get(timeout=10)
             except queue.Empty:
-                if not self.stop_event.is_set():
-                    self.refresh_cache()
                 continue
 
             logger.debug(f"Event received: {event_type} {camera} {event_data['id']}")
-            self.refresh_cache()
 
             if event_type == "start":
                 self.events_in_process[event_data["id"]] = event_data
 
             if event_type == "end":
-                clips_config = self.config.cameras[camera].clips
-
-                clip_created = False
-                if self.should_create_clip(camera, event_data):
-                    if clips_config.enabled and (
-                        clips_config.objects is None
-                        or event_data["label"] in clips_config.objects
-                    ):
-                        clip_created = self.create_clip(
-                            camera,
-                            event_data,
-                            clips_config.pre_capture,
-                            clips_config.post_capture,
-                        )
+                record_config: RecordConfig = self.config.cameras[camera].record
 
-                if clip_created or event_data["has_snapshot"]:
+                has_clip = self.should_create_clip(camera, event_data)
+
+                # Wait for recordings to be ready
+                if has_clip:
```

Comment:
This will store the event with has_clip=False for long events correct? In reality, we could actually have the recorded footage of the event. I know this is in place to ensure the events can send the `clip_ready` mqtt message, but I still feel like we can find a better way.

Reply:
No. This doesn’t get checked until the event ended, and we are just verifying that a recording after the end timestamp has been written. It is no different than the logic that used to exist in refresh cache in practice.